package/MDAnalysis/analysis/psa.py

```diff
@@ -575,7 +575,7 @@ def dist_mat_to_vec(N, i, j):
       int, index (of the matrix element) in the corresponding distance vector
     """
 
-    if not (isinstance(N, int) or isinstance(i, int) or isinstance(j, int)):
+    if not (isinstance(N, numbers.Integral) or isinstance(i, numbers.Integral) or isinstance(j, numbers.Integral)):
```

Comment:
The current check is testing if the input is an integer. In you test you should just make sure that it's ok to use numpy integers like np.int32 in the function. That is important if one wants to use integers from numpy arrays.